app/(protected)/createTodo.tsx

```diff
@@ -0,0 +1,42 @@
+import Button from "@/components/Button";
+import { useState, useContext } from "react";
+import { SafeAreaView, View } from "react-native";
+import CustomTextInput from "@/components/CustomTextInput";
+import { db } from "@/utilities/database";
+import { id } from "@instantdb/react-native";
+import { AuthContext } from "@/utilities/authContext";
+import { useRouter } from "expo-router";
+import DateSelect from "@/components/DateSelect";
+
+export default function CreateTodo() {
+  const [label, setLabel] = useState<string>("");
+  const [date, setDate] = useState<string>("");
+  const { user } = useContext(AuthContext);
+  const router = useRouter();
+
+  const pushTodo = () => {
+    if (!user || !label.trim()) return;
```

Comment:
You might want to make a hook like `useLoggedInUser`, which throws if user doesn't exist. This way you won't have to check for existence in any component that's under the "logged in" tree